### PR TITLE
feat(windows): implement `with_tooltip`

### PR DESCRIPTION
**Pull request by luoffei**
### What kind of change does this PR introduce?
- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?
- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [ ] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme]（https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md）.
- [ ] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

**Review by amrbashir — CHANGES_REQUESTED** [review]
Thank you for the contribution. 

The builder implementation is LGTM but, you also need to add `set_tooltip` on `SystemTrayHandle` structs.

File: core/tauri-runtime-wry/src/system_tray.rs [hunk 1]
```diff
@@ -172,6 +176,17 @@ impl<T: UserEvent> TrayHandle for SystemTrayHandle<T> {
       .map_err(|_| Error::FailedToSendMessage)
   }
 
+  #[cfg(any(windows, target_os = "macos"))]
```
Comment (amrbashir) [inline core/tauri-runtime-wry/src/system_tray.rs]: No need for guard flags here I think, tao's `SystemTray::set_tooltip` exists on Linux but is a no-op.

**Review by amrbashir — CHANGES_REQUESTED** [review]

File: core/tauri-runtime/src/menu.rs [hunk 2]
```diff
@@ -154,6 +154,8 @@ pub trait TrayHandle: fmt::Debug + Clone + Send + Sync {
   fn set_icon_as_template(&self, is_template: bool) -> crate::Result<()>;
   #[cfg(target_os = "macos")]
   fn set_title(&self, title: &str) -> crate::Result<()>;
+  #[cfg(any(windows, target_os = "macos"))]
```
Comment (amrbashir) [inline core/tauri-runtime/src/menu.rs]: this also needs to be removed

File: core/tauri/src/app/tray.rs [hunk 3]
```diff
@@ -64,6 +64,8 @@ pub struct SystemTray {
   icon_as_template_set: bool,
   #[cfg(target_os = "macos")]
   title: Option<String>,
+  #[cfg(any(windows, target_os = "macos"))]
```
Comment (amrbashir) [inline core/tauri/src/app/tray.rs]: also this

File: core/tauri/src/app/tray.rs [hunk 4]
```diff
@@ -98,6 +100,8 @@ impl Default for SystemTray {
       menu_on_left_click_set: false,
       #[cfg(target_os = "macos")]
       title: None,
+      #[cfg(any(windows, target_os = "macos"))]
```
Comment (amrbashir) [inline core/tauri/src/app/tray.rs]: and this

File: core/tauri/src/app/tray.rs [hunk 5]
```diff
@@ -257,6 +261,31 @@ impl SystemTray {
...
+  ///     let mut tray_builder = SystemTray::new();
+  ///     #[cfg(any(windows, target_os = "macos"))]
+  ///     {
+  ///       tray_builder = tray_builder.with_tooltip("My App");
+  ///     }
+  ///     let tray_handle = tray_builder.build(app)?;
+  ///     Ok(())
+  ///   });
+  /// ```
+  #[cfg(any(windows, target_os = "macos"))]
```
Comment (amrbashir) [inline core/tauri/src/app/tray.rs]: and this

File: core/tauri/src/app/tray.rs [hunk 6]
```diff
@@ -414,6 +443,11 @@ impl SystemTray {
       }
     }
 
+    #[cfg(any(windows, target_os = "macos"))]
```
Comment (amrbashir) [inline core/tauri/src/app/tray.rs]: and this

File: core/tauri/src/app/tray.rs [hunk 7]
```diff
@@ -610,6 +644,12 @@ impl<R: Runtime> SystemTrayHandle<R> {
     self.inner.set_title(title).map_err(Into::into)
   }
 
+  /// Adds a tooltip for this tray icon.
+  #[cfg(any(windows, target_os = "macos"))]
```
Comment (amrbashir) [inline core/tauri/src/app/tray.rs]: and this

**Review by amrbashir — CHANGES_REQUESTED** [review]

File: core/tauri/src/app/tray.rs [hunk 8]
```diff
@@ -257,6 +259,25 @@ impl SystemTray {
     self
   }
 
+  /// Sets the tray icon tooltip.
+  ///
```
Comment (amrbashir) [inline core/tauri/src/app/tray.rs:263]:
```suggestion
  ///
  /// ## Platform-specific:
  ///
  /// - **Linux:** Unsupported
  ///
```

File: core/tauri/src/app/tray.rs [hunk 9]
```diff
@@ -610,6 +635,11 @@ impl<R: Runtime> SystemTrayHandle<R> {
     self.inner.set_title(title).map_err(Into::into)
   }
 
+  /// Adds a tooltip for this tray icon.
```
Comment (amrbashir) [inline core/tauri/src/app/tray.rs]:
```suggestion
  /// Set the tooltip for this tray icon.
  ///
  /// ## Platform-specific:
  ///
  /// - **Linux:** Unsupported
```

**Review by amrbashir — CHANGES_REQUESTED** [review]

**Review by amrbashir — DISMISSED** [review]
Thank you ❤️

**Review by lucasfernog — APPROVED** [review]
Nice work, thanks!

**Pull request merged**